extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/JoinResultUpdater.java

```diff
@@ -420,10 +423,20 @@ public VisibilityBindingSet next() {
                 bs.addBinding(binding);
             }
 
+            // We want to make sure the visibilities are always written the same way,
+            // so figure out which are on the left side and which are on the right side.
+            final String leftVisi;
+            final String rightVisi;
+            if(newResultSide == Side.LEFT) {
+                leftVisi = newResult.getVisibility();
+                rightVisi = joinResult.getVisibility();
+            } else {
+                leftVisi = joinResult.getVisibility();
+                rightVisi = newResult.getVisibility();
+            }
+
             String visibility = "";
             final Joiner join = Joiner.on(")&(");
-            final String leftVisi = newResult.getVisibility();
-            final String rightVisi = joinResult.getVisibility();
```

Comment:
Using flatten makes a bunch of unit tests fail, so I'd rather address that in a different ticket.